enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,364 @@
+2. Check if annotation `mtu-migration` has a value, and if so, that it is a
+   valid target host MTU value with respect to the target MTU value set in the
+   operator configuration. If not, consider the MTU change unsafe and don't
+   proceed. Otherwise, consider this to be the target host MTU value or assume a
+   default target host MTU value of the target MTU value + 100.
+3. Render the OVN-K configuration. Given the current `mtu` and `routable-mtu`
+   settings (a `routable-mtu`might be set from a previous MTU migration
+   attempt), and the `target` MTU value:
+   * If `target` > `mtu` > `routable-mtu`, set `mtu` to `target` and keep the
+     current value of `routable-mtu`.
+   * If `mtu` > `target` > `routable-mtu`, set `mtu` to `target` and keep the
+     current value of `routable-mtu`.
+   * If `mtu` > `routable-mtu` > `target`, set `routable-mtu` to `target` and
+     keep the current value of `mtu`.
+   * If `target` > `mtu` and `routable-mtu` is unset, set `mtu` to `target` and
+     `routable-mtu` to the previous value of `mtu`.
+   * If `mtu` > `target` and `routable-mtu` is unset, set `routable-mtu` to
+     `target` and keep the current value for `mtu`.
+4. Render the `cno-set-target-mtu-<TARGET_MTU>-<pool>` MachineConfig object for
+   both the `master` and `worker` machine config pools containing the
+   configure-ovs dropin with the target host MTU value as `TARGET_MTU`.
+5. Wait until MCO has applied the Machine Config object and performed the
+   rolling reboot of all nodes.
+6. Render a OVN-K configuration with `mtu` set to the target MTU value and
+   `routable-mtu` unset.
+7. Render the `cno-unset-target-mtu-<pool>` MachineConfig object for both the
+   `master` and `worker` machine config pools containing the configure-ovs
+   dropin with `TARGET_MTU` unset.
+8. Wait until MCO has applied the Machine Config object and performed the
+   rolling reboot of all nodes.
+9. Set the target MTU value in CNO `applied-cluster` config map. MTU value in in
+   CNO `applied-cluster` config map should stay at the current MTU value until
+   this step.
+
+The CNO status is expected to be derived from MachineConfig pool status
+throughout the procedure since CNO renders MachineConfig objects. The
+MachineConfig pool status should only be taken into account if it shall source a
+MachineConfig object that has been rendered by CNO in the latest reconciliation.
+If that is the case:
+* CNO status should be progressing if MachineConfig pool has not yet sourced a
+  MachineConfig or is otherwise pending for managed nodes to be updated and
+  ready.
+* CNO status should be degraded if MachineConfig pool status is degraded.
+
+The CNO should clean up MachineConfig objects that are no longer needed in the
+procedure which deserves a closer look since this might trigger unwanted
+reboots. This happens at two times in the procedure:
+* `cno-set-target-mtu-<TARGET_MTU>-<pool>` MachineConfig object is removed when
```

Comment:
The MCO doesn't have a batch changes concept. If you really want, you could just modify the existing MachineConfig from A to B (so instead of creating `cno-unset-target-mtu-<pool>`, you can simply modify `cno-set-target-mtu-<TARGET_MTU>-<pool>` to the new desired state

Reply:
@yuqi-zhang I agree that it is preferable to just modify the contents but...

How can we know when a node has been updated with a MachineConfig object with specific contents?? The only way I know is to parse the rendered MachineConfig object and look there for the contents of the source MachineConfig. This is not very practical.

Whereas if we use different MachineConfig objects with deterministic names, we can just check if that name is listed in the source of the MachineConfigPool.